Mathlib/Data/Nat/Sqrt.lean

```diff
@@ -201,4 +201,12 @@ lemma not_exists_sq (hl : m * m < n) (hr : n < (m + 1) * (m + 1)) : ¬∃ t, t *
 lemma not_exists_sq' : m ^ 2 < n → n < (m + 1) ^ 2 → ¬∃ t, t ^ 2 = n := by
   simpa only [Nat.pow_two] using not_exists_sq
 
+lemma le_sqrt_of_eq_mul {a b c : ℕ} (h : a = b * c) : b ≤ a.sqrt ∨ c ≤ a.sqrt :=
+  if hle : b ≤ a.sqrt then
+    Or.inl hle
+  else
+    Or.inr (Nat.le_of_lt_succ (Nat.lt_of_mul_lt_mul_left (lt_of_lt_of_le (h ▸ Nat.lt_succ_sqrt a)
```

Comment:
Providing a two-line proof term will not be that easy to modify or maintain. You're free to use a tactic!
`right; nlinarith [Nat.lt_succ_sqrt' a]` shoulds like a perfectly fine proof.